packages/frontend/src/components/Game/GameBoard/index.jsx

```diff
@@ -1,17 +1,26 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import useGameContext from 'hooks/useGameContext';
 import ScoreBoard from '../ScoreBoard';
 import TileGrid from './TileGrid';
-
+import { handleGameOver } from 'contexts/game/actions';
+import ResultDisplay from './ResultDisplay';
 const GameBoard = () => {
   const {
-    state: { players, currentPlayer, tiles },
+    state: { players, currentPlayer, tiles, stage },
+    dispatch,
   } = useGameContext();
 
+  useEffect(() => {
+    const matchedTiles = tiles.filter((tile) => tile.isMatched === true);
+    if (tiles.length !== 0 && matchedTiles.length === tiles.length) {
+      dispatch(handleGameOver());
+    }
```

Comment:
I feel like the `handleGameOver` action should be dispatched when the game stage is `GAME_STAGES.GAME_OVER`. This `useEffect` could be in the `Game` component and handle all the stages with a switch or if/else